src/lib.rs

```diff
@@ -815,6 +954,72 @@ impl Location {
     }
 }
 
+/// A builder for [`Location`](struct.Location.html).
+///
+/// Often used with [`RecordBuilder`](struct.RecordBuilder.html)'s [`new`], which
+/// takes a `&Location`.
+///
+/// # Example
+///
+/// ```rust
+/// use log::LocationBuilder;
+///
+/// let error_location = LocationBuilder::new()
+///                         .module_path("/path/to/module")
+///                         .file("lib.rs")
+///                         .line(100)
+///                         .build();
+/// ```
+/// [`new`]: struct.RecordBuilder.html#method.new
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct LocationBuilder {
+    location: Location
+}
+
+impl LocationBuilder {
+    /// Construct a new `LocationBuilder`.
+    ///
+    /// The default options are:
+    ///
+    /// - `module_path`: [`module_path!()`]
+    /// - `file`: [`file!()`]
+    /// - `line`: 0
+    /// [`module_path!()`]: https://doc.rust-lang.org/std/macro.module_path.html
+    /// [`file!()`]: https://doc.rust-lang.org/std/macro.file.html
+    pub fn new() -> LocationBuilder {
+        LocationBuilder {
+            location: Location {
+                __module_path: module_path!(),
```

Comment:
These defaults are not going to be particularly useful - I'd probably lean towards making them empty strings.